Arrowgene.Ddon.Shared/Model/Storages.cs

```diff
@@ -119,6 +119,13 @@ private uint DeterminePawnId(Character character, StorageType storageType, ushor
                 return 0;
             }
         }
+
+        public static ushort DeterminePawnEquipSlot(ushort storageSlot)
+        {
+            int pawnIndex = storageSlot / (EquipmentTemplate.TOTAL_EQUIP_SLOTS * 2);
```

Comment:
Should this be `storageSlot - 1`?

Reply:
No. In the pawn equipment storage, slots from 1 to 15 are pawn 1 perf. equipment, from 16 to 30 are pawn 1 visual equipment, 31 to 45 are pawn 2 perf. equipment, and so on.

Reply:
Ah, true, this means that the function above also has it wrong, since slot 30 is for pawn of index 0, pawnIndex = (30 - 1)/30 = 0, otherwise, slot 30 would be reported as being for pawn of index 1.
You should probably merge the logic here with GetEquipTypeFromSlotNo so you have a function that returns the pawn index, the equipment slot and the equip type

Reply:
In the context of equipment, visual equip is equip slots 1 to 15. In the context of the storage that holds its items, visual equip is storage slots 16 to 30 (or 46 to 60 in case of pawn 2, etc.)

Reply:
So the idea of anything over 15 is purely a server side concept? or is this also how capcom did it?

Reply:
The client gets this information when we send the client the storage items, and it seems to work, as the client wants to get the equipment as storage type CharacterEquipment, whether capcom did it like we did or not, with the slots in a different order, is something i don't know. I just made the equip slots map to storage slots like this for ease of use and the client seems happy with it